src/validators/dataclass.rs

```diff
@@ -441,6 +442,17 @@ impl BuildValidator for DataclassValidator {
             None
         };
 
+        let slots = match schema.get_as::<&PyList>(intern!(py, "slots"))? {
```

Comment:
Are we implicitly making the assumption that dataclasses are all-slots or no-slots-at-all? (Just want to make sure things work correctly since we just confirmed that's not the case, since you can add non-slots fields in subclasses of slots dataclasses.)

Seems to me if you always take the `__dict__` (if present) _and_  grab the `__slots__`, things should work, but the of course the devil is in the implementation details.